wasp/src/main/java/com/orhanobut/wasp/MockFactory.java

```diff
@@ -17,32 +20,52 @@
 
     private static MockFactory mockFactory;
 
-    private MockFactory() {
+    private Context context;
+
+    private MockFactory(final Context context) {
         // no instance
+        this.context = context;
     }
 
-    static MockFactory getDefault() {
+    static MockFactory getDefault(final Context context) {
         if (mockFactory == null) {
-            mockFactory = new MockFactory();
+            mockFactory = new MockFactory(context);
         }
         return mockFactory;
     }
 
     @Override
     public <T> void invokeRequest(WaspRequest waspRequest, CallBack<T> callBack) {
-        MockType mockType = waspRequest.getMockType();
+        WaspMock mock = waspRequest.getMock();
+        int statusCode = mock.getStatusCode();
 
-        if (mockType == MockType.FAIL) {
-            callBack.onError(new WaspError("mock url", "Mock test fail", 404));
+        if (statusCode < 200 || statusCode > 299) {
+            callBack.onError(new WaspError("mock url", "Mock test fail", statusCode));
             return;
         }
 
-        //Create mock object and return
-        MethodInfo methodInfo = waspRequest.getMethodInfo();
-        Type responseType = methodInfo.getResponseObjectType();
-        String responseString = createJsonString(responseType);
+        String responseString = null;
+
+        if (mock.getPath().equals("")) {
+
+            //Create mock object and return
+            MethodInfo methodInfo = waspRequest.getMethodInfo();
+            Type responseType = methodInfo.getResponseObjectType();
+            responseString = createJsonString(responseType);
+
+        } else {
+
+            try {
+                responseString = IOUtils.readFileFromAssets(context, mock.getPath());
+            } catch (IOException e) {
+                callBack.onError(new WaspError("mock url", "Mocking failed due to file io error.",
```

Comment:
We can use group formatting here
<pre>
callBack.onError(new WaspError(
          "mock url", 
          "Mocking failed due to file io error.",
          WaspError.INVALID_STATUS_CODE
));
</pre>